stbt_lint.py

```diff
@@ -77,7 +76,8 @@ def main(argv):
         stderr=subprocess.PIPE)
 
     t = threading.Thread(target=filter_warnings,
-                         args=(pylint.stderr, sys.stderr))
+                         args=(pylint.stderr,
+                               os.fdopen(sys.stderr.fileno(), "wb")))
```

Comment:
>Side-effect of this change: On Python3, the stderr output will be
>buffered. Line-buffering can only be set in text mode. On Python 2 the
>output is still line-buffered.

Perhaps set it to unbuffered.  We're reading a line at a time, so will be writing a line at a time too.  It would effectively be line-buffered.

```suggestion
                               os.fdopen(sys.stderr.fileno(), "wb", 0)))
```

Reply:
Is that going to get intermingled with anything that pylint prints to stdout?

Reply:
Fixed in #594.

Reply:
No - at least in the absence of very long lines - because stdout is line buffered, so all calls to `write` (the syscall) will end in a `\n`, and the same is true for stderr because we're calling `readline`.